### PR TITLE
Fixed https://issues.jenkins-ci.org/browse/JENKINS-55955

### PR DESCRIPTION
**Pull request by mrinal1209**
Remove JAXB dependency from notification plugin

@oleg-nenashev @martinda  please check if I have resolved the needed.

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
CC @jenkinsci/java11-support

**Comment by martinda** [comment]
I think you want to replace the decoder, not remove it. Have a look at the [Base64](https://docs.oracle.com/javase/10/docs/api/java/util/Base64.html) class.

**Comment by mrinal1209** [comment]
@MRamonLeon  and @martinda  I have tried to keep the similar of current behavior.
https://stackoverflow.com/questions/16669723/whats-the-difference-between-new-stringbyte-and-datatypeconverter-printbase

File: src/main/java/com/tikal/hudson/plugins/notification/Protocol.java [hunk 1]
```diff
@@ -78,7 +77,7 @@ protected void send(String url, byte[] data, int timeout, boolean isJson) throws
             connection.setRequestProperty("Content-Type", String.format( "application/%s;charset=UTF-8", isJson ? "json" : "xml" ));
             String userInfo = targetUrl.getUserInfo();
             if (null != userInfo) {
-              String b64UserInfo = DatatypeConverter.printBase64Binary(userInfo.getBytes());
+              String b64UserInfo = new String(userInfo.getBytes());
```
Comment (batmat) [inline src/main/java/com/tikal/hudson/plugins/notification/Protocol.java]: This is definitely changing the behavior. 
You likely want to re-read this code, and read about https://en.wikipedia.org/wiki/Base64

Probably @mrinal1209 you want to test this on your IDE or so, and basically print to stdout for instance to understand the difference between the previous code, and the change you are proposing.

**Comment by batmat** [comment]
> I have tried to keep the similar of current behavior.

@mrinal1209 that example you're linking is showing some code trying to encode as base64 a string that is *already* encoded as such. 
Here we have bytes that the code was encoding in base64, which would not be the case anymore with your change.

**Comment by mrinal1209** [comment]
@batmat Okay I will read more and make the following changes needed . This my first time GSOC and open source so I am learning new stuff everyday so very sorry if I mess up :)

**Comment by mrinal1209** [comment]
can any one tell me why I am getting this error ?
[ERROR] /home/jenkins/workspace/lugins_notification-plugin_PR-36/src/main/java/com/tikal/hudson/plugins/notification/Protocol.java:81: Undefined reference: java.util.Base64.Encoder java.util.Base64.getEncoder()

@oleg-nenashev @martinda  @batmat @MRamonLeon

**Comment by MRamonLeon** [comment]
@mrinal1209 It's because animal-sniffer is using signature java17. It's because the puglin dependency is too old and is requiring java 7, but the new method used is available on Java 8. I think that the best approach is to update the parent dependence of your pom to a newest org.jenkins-ci..plugins:plugin version.

**Comment by mrinal1209** [comment]
@MRamonLeon  what changes should I make now I am not able to proceed .
If I make changes here in
   <parent>
        <groupId>org.jenkins-ci.plugins</groupId>
        <artifactId>plugin</artifactId>
        <version>2.9</version> // change it to latest version to 3.37
</parent>
my error is resolving but still building is failing for other reasons.
Could not resolve dependencies for project com.tikalk.hudson.plugins:notification:hpi:1.14-SNAPSHOT: Could not find artifact org.jenkins-ci.main:jenkins-war:executable-war:2.60.1 in repo.jenkins-ci.org (http://repo.jenkins-ci.org/public/)

**Comment by mrinal1209** [comment]
@martinda @oleg-nenashev @batmat @MRamonLeon 
this bug is taking more than the expected time please any one can help me ? I am not able to understand or make changes to POM file whatever I have tried is failing to build the project . Please help..

**Comment by mrinal1209** [comment]
Out of my curiosity just for test took a git clone from the original repository  for notification plugin and applied mvn clean install it is giving me this error :- 
**Failed to execute goal org.codehaus.mojo:findbugs-maven-plugin:3.0.3:findbugs (findbugs) on project notification: Unable to parse configuration of mojo org.codehaus.mojo:findbugs-maven-plugin:3.0.3:findbugs for parameter pluginArtifacts: Cannot assign configuration entry 'pluginArtifacts' with value '${plugin.artifacts}' of type java.util.Collections.UnmodifiableRandomAccessList to property of type java.util.ArrayList** 

@oleg-nenashev @martinda @batmat @MRamonLeon  why is this so ?  if master branch is not working properly how could I ever able to fix the bug in the first place ? Please respond

**Comment by mrinal1209** [comment]
**[ERROR] Failed to execute goal on project notification: Could not resolve dependencies for project com.tikalk.hudson.plugins:notification:hpi:1.14-SNAPSHOT: Could not find artifact org.jenkins-ci.main:jenkins-war:executable-war:1.642.3 in azure-incrementals (https://repo.azure.jenkins.io/incrementals/)** 


Still I am having this error if someone can help @AniketSK

**Comment by batmat** [comment]
Not on my laptop right now, but pretty sure https://github.com/jenkinsci/notification-plugin/pull/36#issuecomment-467619909 is an issue with Maven version. I think you're using 3.6.0, try using 3.5.0 instead, or the other way round.

**Comment by martinda** [comment]
I run into dependencies issues as well building this pull-request branch. I tried maven 3.3.9, 3.5.0, 3.5.4, and 3.6.0. I do not know what else to try. My command line is:
```
mvn clean test -U
```
The error is:
```
Could not resolve dependencies for project com.tikalk.hudson.plugins:notification:hpi:1.14-SNAPSHOT: Could not find artifact org.jenkins-ci.main:jenkins-war:executable-war:1.642.3 in repo.jenkins-ci.org (http://repo.jenkins-ci.org/public/)
```

**Comment by mrinal1209** [comment]
> I run into dependencies issues as well building this pull-request branch. I tried maven 3.3.9, 3.5.0, 3.5.4, and 3.6.0. I do not know what else to try. My command line is:
> 
> ```
> mvn clean test -U
> ```
> 
> The error is:
> 
> ```
> Could not resolve dependencies for project com.tikalk.hudson.plugins:notification:hpi:1.14-SNAPSHOT: Could not find artifact org.jenkins-ci.main:jenkins-war:executable-war:1.642.3 in repo.jenkins-ci.org (http://repo.jenkins-ci.org/public/)
> ```

@martinda  and @batmat  Thanks for taking your time out please help me with this problem .

File: pom.xml [hunk 2]
```diff
@@ -9,13 +16,15 @@
...
-    <parent>
-        <groupId>org.jenkins-ci.plugins</groupId>
-        <artifactId>plugin</artifactId>
-        <version>2.9</version>
-    </parent>
-
     <properties>
+        <jenkins.version>1.642.3</jenkins.version>
+        <java.level>8</java.level>
+        <maven-surefire-plugin.version>2.19.1</maven-surefire-plugin.version>
```
Comment (oleg-nenashev) [inline pom.xml]: This entry and ones below should not be needed

**Comment by oleg-nenashev** [comment]
Hmm... it was not integrated so far. @mrinal1209 are you still interested? I am not sure what's the status, just got a notification

**Comment by mrinal1209** [comment]
@oleg-nenashev  not now little bit caught up with work I'll start contributing after GSoC ends.

**Pull request closed without merging**